File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        map<char,int> lets;
        for(char x: s) lets[x]++;
        for(int i=0;i<s.length();i++) if(lets[s[i]] == 1) return i;
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = {}
        for i in range(len(s)):
            if s[i] not in freq:
                freq[s[i]] = []
            freq[s[i]].append(i)
        for c in s:
            if len(freq[c]) == 1:
                return freq[c][0]
        return -1